src/lib/cozy-client/adapters/PouchdbAdapter.js

```diff
@@ -38,21 +44,45 @@ export default class PouchdbAdapter {
       .then(result => result.update_seq)
   }
 
-  async sync(dispatch, direction = SYNC_BIDIRECTIONAL) {
-    const baseUrl = await this.getReplicationBaseUrl()
-    for (let doctype of this.doctypes) {
+  async startSync(dispatch, direction = SYNC_BIDIRECTIONAL) {
+    try {
+      const infos = await this.sync(dispatch, direction)
+      this.scheduleNextSync(dispatch, direction)
+      return infos
+    } catch (err) {
+      this.scheduleNextSync(dispatch, direction)
+      throw err
+    }
```

Comment:
You could use a `finally` block here to ` this.scheduleNextSync(dispatch, direction)`